### PR TITLE
feature/#71-new-finding-azure-sql-services

### PR DESCRIPTION
**Pull request by misg**
This PR implements all the CIS findings related to the SQL Database service (and mentionned in https://github.com/nccgroup/ScoutSuite-Proprietary/issues/71).

### CODE REVIEW

**Comment by j4v** [comment]
Just checking but this is merging some of the refactoring into `develop`, no?

**Comment by misg** [comment]
> Just checking but this is merging some of the refactoring into `develop`, no?

I just changed the base 30sec ago, you're too fast. :p

File: ScoutSuite/providers/azure/rules/findings/sqldatabase-databases-no-auditing.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
     "dashboard_name": "SQL Databases",
-    "description": "Auditing disabled",
-    "rationale": "You should enable auditing for all of your SQL databases. See CIS 4.2.1.",
+    "description": "Auditing is disabled for some SQL databases",
```
Comment (j4v) [inline ScoutSuite/providers/azure/rules/findings/sqldatabase-databases-no-auditing.json]: `Auditing disabled for SQL databases` would be more in line with current descriptions

File: ScoutSuite/providers/azure/rules/findings/sqldatabase-databases-no-threat-detection.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
     "dashboard_name": "SQL Databases",
-    "description": "Threat detection disabled",
-    "rationale": "You should enable threat detection for all of your SQL databases. See CIS 4.2.2.",
+    "description": "Threat detection is disabled for some SQL databases",
```
Comment (j4v) [inline ScoutSuite/providers/azure/rules/findings/sqldatabase-databases-no-threat-detection.json]: `Theat detection disabled for SQL databases`

**Comment by j4v** [comment]
moooooar findings :clap: :clap: :clap:

File: ScoutSuite/providers/azure/resources/resources.py [hunk 3]
```diff
@@ -3,10 +3,14 @@
 from ScoutSuite.providers.base.configs.resources import CompositeResources
 
 
+# TODO: add docstrings.
 class AzureCompositeResources(CompositeResources, metaclass=abc.ABCMeta):
 
     async def _fetch_children(self, parent, **kwargs):
-        for child_class in self._children:
+        for child_class, name in self._children:
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/resources.py]: Doesn't it throw a not enough values to unpack kind of error if there is no name?

Comment (Aboisier) [reply ScoutSuite/providers/azure/resources/resources.py]: Oh, I see, you put 'None'.

**Pull request merged**